### Nama : Rini Wulandari

### NIM : 2409116048

### Kelas : SI B 2024

##**BUSINESS UNDERSTANDING**

###**Business Objective**

Tujuan utama dari proyek ini adalah untuk meningkatkan pengalaman pelanggan dan kinerja platform pemesanan makanan online. Secara lebih spesifik dengan berusaha meningkatkan kepuasan pelanggan, meningkatkan layanan pengiriman dengan menganalisis umpan balik pelanggan dan status pesan, dan menggunakan data demografi dan preferensi pelanggan untuk menetapkan harga yang lebih relevan.

###**Assess Situation**

Pada dataset ini memiliki kekuatan dengan informasi demografis yang lengkap dan umpan balik pelanggan yang memberikan wawasan tentang kepuasan dan kualitas layanan. Namun, ada keterbatasan seperti tidak adanya data tentang preferensi makanan atau interaksi pelanggan sebelumnya. Meskipun ada peluang untuk meningkatkan pengalaman pelanggan dengan rekomendasi produk yang disesuaikan berdasarkan demografi. Selain itu ancaman yang perlu diperhatikan adalah potensi kesalahan dalam analisis data yang dapat menghasilkan rekomendasi yang tidak akurat, serta perubahan preferensi pelanggan atau faktor eksternal lainnya yang dapat memengaruhi hasil analisis.

###**Data Mining Goals**

Tujuan ini meliputi segmentasi pelanggan berdasarkan atribut demografis untuk menyesuaikan strategi pemasaran dan promosi, analisis kepuasan pelanggan dengan mengidentifikasi faktor-faktor yang mempengaruhi kepuasan, serta membangun model prediksi untuk status pesanan. Selain itu, tujuan lainnya adalah mengenali pola pembelian berdasarkan lokasi dan waktu, serta menganalisis data geografis untuk meningkatkan distribusi dan logistik pengiriman.

###**Project Plan**

Pada project plan ada beberapa tahapan yang bisa dilakukan yaitu, pertama dapat dilakukan dengan mengumpulkan dataset kemudian  dibersihkan untuk memastikan tidak ada nilai hilang atau duplikat dan memastikan konsistensi data. Kedua akan fokus pada eksplorasi data melalui analisis statistik dasar, korelasi antar atribut, dan visualisasi untuk mendapatkan wawasan lebih mendalam. Ketiga melibatkan pengembangan model data mining untuk memahami pola pembelian. Keempat, model akan dievaluasi dan divalidasi menggunakan metrik yang sesuai dan data uji untuk memastikan keakuratan dan konsistensi hasil. Terakhir, penyusunan laporan hasil analisis dan rekomendasi.

Setiap langkah-langkah dalam proses ini dirancang untuk memastikan data yang akurat dan model yang dapat memberikan wawasan serta rekomendasi yang berguna bagi pelanggan.

##**DATA UNDERSTANDING**

In [46]:
import pandas as pd

In [47]:
df = pd.read_csv("/content/onlinefoods.csv")
df

Age  Gender Marital Status Occupation  Monthly Income  \
0     20  Female         Single    Student       No Income   
1     24  Female         Single    Student  Below Rs.10000   
2     22    Male         Single    Student  Below Rs.10000   
3     22  Female         Single    Student       No Income   
4     22    Male         Single    Student  Below Rs.10000   
..   ...     ...            ...        ...             ...   
383   23  Female         Single    Student       No Income   
384   23  Female         Single    Student       No Income   
385   22  Female         Single    Student       No Income   
386   23    Male         Single    Student  Below Rs.10000   
387   23    Male         Single    Student       No Income   

    Educational Qualifications  Family size  latitude  longitude  Pin code  \
0                Post Graduate            4   12.9766    77.5993    560001   
1                     Graduate            3   12.9770    77.5773    560009   
2                Post Graduate            3   12.9551    77.6593    560017   
3                     Graduate            6   12.9473    77.5616    560019   
4                Post Graduate            4   12.9850    77.5533    560010   
..                         ...          ...       ...        ...       ...   
383              Post Graduate            2   12.9766    77.5993    560001   
384              Post Graduate            4   12.9854    77.7081    560048   
385              Post Graduate            5   12.9850    77.5533    560010   
386              Post Graduate            2   12.9770    77.5773    560009   
387              Post Graduate            5   12.8988    77.5764    560078   

    Output   Feedback Unnamed: 12  
0      Yes   Positive         Yes  
1      Yes   Positive         Yes  
2      Yes  Negative          Yes  
3      Yes   Positive         Yes  
4      Yes   Positive         Yes  
..     ...        ...         ...  
383    Yes   Positive         Yes  
384    Yes   Positive         Yes  
385    Yes   Positive         Yes  
386    Yes   Positive         Yes  
387    Yes   Positive         Yes  

[388 rows x 13 columns]

###**Pengecekan Struktur Data**

In [48]:
print("Menampilkan Beberapa Row Data :")
print(df.head())

Menampilkan Beberapa Row Data :
   Age  Gender Marital Status Occupation  Monthly Income  \
0   20  Female         Single    Student       No Income   
1   24  Female         Single    Student  Below Rs.10000   
2   22    Male         Single    Student  Below Rs.10000   
3   22  Female         Single    Student       No Income   
4   22    Male         Single    Student  Below Rs.10000   

  Educational Qualifications  Family size  latitude  longitude  Pin code  \
0              Post Graduate            4   12.9766    77.5993    560001   
1                   Graduate            3   12.9770    77.5773    560009   
2              Post Graduate            3   12.9551    77.6593    560017   
3                   Graduate            6   12.9473    77.5616    560019   
4              Post Graduate            4   12.9850    77.5533    560010   

  Output   Feedback Unnamed: 12  
0    Yes   Positive         Yes  
1    Yes   Positive         Yes  
2    Yes  Negative          Yes  
3    Yes   Pos

In [49]:
print("Menampilkan Info Data :")
print(df.info())

Menampilkan Info Data :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 388 entries, 0 to 387
Data columns (total 13 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Age                         388 non-null    int64  
 1   Gender                      388 non-null    object 
 2   Marital Status              388 non-null    object 
 3   Occupation                  388 non-null    object 
 4   Monthly Income              388 non-null    object 
 5   Educational Qualifications  388 non-null    object 
 6   Family size                 388 non-null    int64  
 7   latitude                    388 non-null    float64
 8   longitude                   388 non-null    float64
 9   Pin code                    388 non-null    int64  
 10  Output                      388 non-null    object 
 11  Feedback                    388 non-null    object 
 12  Unnamed: 12                 388 non-null    object 
dtypes: float64(

In [50]:
print("Menampilkan Jumlah Baris dan Kolom :")
print(df.shape)

Menampilkan Jumlah Baris dan Kolom :
(388, 13)


In [51]:
print("Menampilkan Tipe Data Setiap Kolom :")
print(df.dtypes)

Menampilkan Tipe Data Setiap Kolom :
Age                             int64
Gender                         object
Marital Status                 object
Occupation                     object
Monthly Income                 object
Educational Qualifications     object
Family size                     int64
latitude                      float64
longitude                     float64
Pin code                        int64
Output                         object
Feedback                       object
Unnamed: 12                    object
dtype: object


###**Statistik Deskriptif**

In [52]:
print("\nMenampilkan Statistik Deskriptif :")
print(df.describe())


Menampilkan Statistik Deskriptif :
              Age  Family size    latitude   longitude       Pin code
count  388.000000   388.000000  388.000000  388.000000     388.000000
mean    24.628866     3.280928   12.972058   77.600160  560040.113402
std      2.975593     1.351025    0.044489    0.051354      31.399609
min     18.000000     1.000000   12.865200   77.484200  560001.000000
25%     23.000000     2.000000   12.936900   77.565275  560010.750000
50%     24.000000     3.000000   12.977000   77.592100  560033.500000
75%     26.000000     4.000000   12.997025   77.630900  560068.000000
max     33.000000     6.000000   13.102000   77.758200  560109.000000


In [53]:
kolom_tertentu = "Gender"
if kolom_tertentu in df.columns:
    print(f"\nMenghitung jumlah kemunculan nilai unik dalam Kolom '{kolom_tertentu}' dengan df['{kolom_tertentu}'].value_counts():")
    print(df[kolom_tertentu].value_counts())
else:
    print(f"Kolom '{kolom_tertentu}' tidak ditemukan dalam DataFrame.")


Menghitung jumlah kemunculan nilai unik dalam Kolom 'Gender' dengan df['Gender'].value_counts():
Gender
Male      222
Female    166
Name: count, dtype: int64


In [55]:
print("\nMenghitung rata-rata :")
print(df.mean(numeric_only=True))


Menghitung rata-rata :
Age                24.628866
Family size         3.280928
latitude           12.972058
longitude          77.600160
Pin code       560040.113402
dtype: float64


In [56]:
print("\nMenghitung median :")
print(df.median(numeric_only=True))


Menghitung median :
Age                24.0000
Family size         3.0000
latitude           12.9770
longitude          77.5921
Pin code       560033.5000
dtype: float64


In [57]:
print("\nMenghitung standar deviasi :")
print(df.std(numeric_only=True))


Menghitung standar deviasi :
Age             2.975593
Family size     1.351025
latitude        0.044489
longitude       0.051354
Pin code       31.399609
dtype: float64


In [58]:
print("\nMenghitung korelasi antar kolom numerik :")
print(df.corr(numeric_only=True))


Menghitung korelasi antar kolom numerik :
                  Age  Family size  latitude  longitude  Pin code
Age          1.000000     0.169982  0.006595   0.047500  0.137294
Family size  0.169982     1.000000 -0.053537   0.070126 -0.009402
latitude     0.006595    -0.053537  1.000000  -0.144234 -0.201813
longitude    0.047500     0.070126 -0.144234   1.000000  0.156119
Pin code     0.137294    -0.009402 -0.201813   0.156119  1.000000


##**DATA PREPARATION**

###**Pengecekan Missing Values**

In [59]:
print((df.isna().sum() / len(df)) * 100)

Age                           0.0
Gender                        0.0
Marital Status                0.0
Occupation                    0.0
Monthly Income                0.0
Educational Qualifications    0.0
Family size                   0.0
latitude                      0.0
longitude                     0.0
Pin code                      0.0
Output                        0.0
Feedback                      0.0
Unnamed: 12                   0.0
dtype: float64


Pada data ini tidak memiliki missing value.

###**Duplicated Values**

In [80]:
df[df.duplicated()]

Empty DataFrame
Columns: [Age, Gender, Marital Status, Occupation, Monthly Income, Educational Qualifications, Family size, latitude, longitude, Pin code, Output, Feedback, Preferred Food Type, Average Delivery Time]
Index: []

In [82]:
df = df.drop_duplicates()

In [83]:
df

Age  Gender Marital Status Occupation   Monthly Income  \
0    20.0  Female         Single    Student        No Income   
1    24.0  Female         Single    Student   Below Rs.10000   
2    22.0    Male         Single    Student   Below Rs.10000   
3    22.0  Female         Single    Student        No Income   
4    22.0    Male         Single    Student   Below Rs.10000   
..    ...     ...            ...        ...              ...   
355  21.0    Male         Single    Student        No Income   
363  30.5    Male        Married   Employee  More than 50000   
369  30.0    Male        Married   Employee  More than 50000   
374  21.0    Male         Single    Student        No Income   
386  23.0    Male         Single    Student   Below Rs.10000   

    Educational Qualifications  Family size  latitude  longitude  Pin code  \
0                Post Graduate            4   12.9766    77.5993    560001   
1                     Graduate            3   12.9770    77.5773    560009   
2                Post Graduate            3   12.9551    77.6593    560017   
3                     Graduate            6   12.9473    77.5616    560019   
4                Post Graduate            4   12.9850    77.5533    560010   
..                         ...          ...       ...        ...       ...   
355                   Graduate            2   13.0012    77.5995    560046   
363                       Ph.D            3   12.9306    77.5434    560085   
369              Post Graduate            6   12.9369    77.6407    560095   
374                   Graduate            3   13.0641    77.5931    560092   
386              Post Graduate            2   12.9770    77.5773    560009   

    Output   Feedback Preferred Food Type  Average Delivery Time  
0      Yes   Positive           Fast Food                     51  
1      Yes   Positive        Healthy Food                     49  
2      Yes  Negative                Vegan                     50  
3      Yes   Positive               Vegan                     27  
4      Yes   Positive           Fast Food                     55  
..     ...        ...                 ...                    ...  
355     No   Positive           Fast Food                     35  
363    Yes   Positive           Fast Food                     25  
369    Yes   Positive      Non-Vegetarian                     53  
374    Yes  Negative                Vegan                     48  
386    Yes   Positive               Vegan                     31  

[285 rows x 14 columns]

Setelah dilakukan drop duplicated, data yang tersisa adalah 285 rows.

###**Outliers**

In [63]:
results = []

cols = df.select_dtypes(include=['float64', 'int64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

# Dataframe dari list hasil
results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

# Tampilkan dataframe
display(results_df)

Kolom        Persentase Outliers
Age                     7.368421
Family size             0.000000
latitude                1.052632
longitude               4.561404
Pin code                0.000000

In [64]:
df

Age  Gender Marital Status Occupation   Monthly Income  \
0     20  Female         Single    Student        No Income   
1     24  Female         Single    Student   Below Rs.10000   
2     22    Male         Single    Student   Below Rs.10000   
3     22  Female         Single    Student        No Income   
4     22    Male         Single    Student   Below Rs.10000   
..   ...     ...            ...        ...              ...   
355   21    Male         Single    Student        No Income   
363   31    Male        Married   Employee  More than 50000   
369   30    Male        Married   Employee  More than 50000   
374   21    Male         Single    Student        No Income   
386   23    Male         Single    Student   Below Rs.10000   

    Educational Qualifications  Family size  latitude  longitude  Pin code  \
0                Post Graduate            4   12.9766    77.5993    560001   
1                     Graduate            3   12.9770    77.5773    560009   
2                Post Graduate            3   12.9551    77.6593    560017   
3                     Graduate            6   12.9473    77.5616    560019   
4                Post Graduate            4   12.9850    77.5533    560010   
..                         ...          ...       ...        ...       ...   
355                   Graduate            2   13.0012    77.5995    560046   
363                       Ph.D            3   12.9306    77.5434    560085   
369              Post Graduate            6   12.9369    77.6407    560095   
374                   Graduate            3   13.0641    77.5931    560092   
386              Post Graduate            2   12.9770    77.5773    560009   

    Output   Feedback Unnamed: 12  
0      Yes   Positive         Yes  
1      Yes   Positive         Yes  
2      Yes  Negative          Yes  
3      Yes   Positive         Yes  
4      Yes   Positive         Yes  
..     ...        ...         ...  
355     No   Positive          No  
363    Yes   Positive         Yes  
369    Yes   Positive         Yes  
374    Yes  Negative          Yes  
386    Yes   Positive         Yes  

[285 rows x 13 columns]

Melakukan Imputasi pada kolom 'Age',
 'Latitude', 'Longitude'.

In [65]:
columns_to_impute = ["Age", "latitude", "longitude"]

for col in columns_to_impute:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Menggunakan .loc[] agar tidak muncul SettingWithCopyWarning
    df.loc[:, col] = df[col].clip(lower=lower_bound, upper=upper_bound)

<ipython-input-65-fb0a47440d2c>:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '[20.  24.  22.  22.  22.  27.  22.  24.  23.  23.  22.  23.  23.  21.
 23.  24.  28.  23.  25.  21.  24.  22.  22.  23.  21.  25.  22.  22.
 23.  22.  22.  22.  25.  22.  22.  25.  25.  30.5 23.  23.  30.  23.
 23.  22.  27.  24.  23.  23.  23.  25.  22.  28.  22.  24.  30.5 25.
 22.  23.  23.  23.  25.  24.  23.  24.  22.  24.  25.  23.  26.  24.
 26.  22.  24.  24.  23.  23.  22.  23.  24.  25.  28.  27.  26.  22.
 23.  26.  30.5 23.  22.  24.  26.  28.  26.  25.  25.  18.5 21.  25.
 23.  25.  30.5 24.  30.5 25.  27.  26.  26.  30.5 24.  27.  25.  28.
 30.5 23.  19.  19.  27.  25.  30.5 26.  22.  23.  22.  25.  30.  24.
 25.  23.  24.  30.5 22.  20.  21.  24.  25.  30.5 27.  20.  21.  26.
 25.  26.  26.  27.  27.  24.  25.  20.  22.  26.  27.  25.  24.  23.
 22.  26.  26.  25.  29.  23.  22.  22.  30.5 28.  25.  26.  30.5 24.
 24.  

Pengecekan Ulang

In [66]:
results = []

cols = df.select_dtypes(include=['float64', 'int64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

# Dataframe dari list hasil
results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

# Tampilkan dataframe
display(results_df)

Kolom        Persentase Outliers
Age                          0.0
Family size                  0.0
latitude                     0.0
longitude                    0.0
Pin code                     0.0

In [73]:
df

Age  Gender Marital Status Occupation   Monthly Income  \
0    20.0  Female         Single    Student        No Income   
1    24.0  Female         Single    Student   Below Rs.10000   
2    22.0    Male         Single    Student   Below Rs.10000   
3    22.0  Female         Single    Student        No Income   
4    22.0    Male         Single    Student   Below Rs.10000   
..    ...     ...            ...        ...              ...   
355  21.0    Male         Single    Student        No Income   
363  30.5    Male        Married   Employee  More than 50000   
369  30.0    Male        Married   Employee  More than 50000   
374  21.0    Male         Single    Student        No Income   
386  23.0    Male         Single    Student   Below Rs.10000   

    Educational Qualifications  Family size  latitude  longitude  Pin code  \
0                Post Graduate            4   12.9766    77.5993    560001   
1                     Graduate            3   12.9770    77.5773    560009   
2                Post Graduate            3   12.9551    77.6593    560017   
3                     Graduate            6   12.9473    77.5616    560019   
4                Post Graduate            4   12.9850    77.5533    560010   
..                         ...          ...       ...        ...       ...   
355                   Graduate            2   13.0012    77.5995    560046   
363                       Ph.D            3   12.9306    77.5434    560085   
369              Post Graduate            6   12.9369    77.6407    560095   
374                   Graduate            3   13.0641    77.5931    560092   
386              Post Graduate            2   12.9770    77.5773    560009   

    Output   Feedback Unnamed: 12 Preferred Food Type  Average Delivery Time  
0      Yes   Positive         Yes            Desserts                     29  
1      Yes   Positive         Yes               Vegan                     51  
2      Yes  Negative          Yes      Non-Vegetarian                     31  
3      Yes   Positive         Yes           Fast Food                     32  
4      Yes   Positive         Yes        Healthy Food                     55  
..     ...        ...         ...                 ...                    ...  
355     No   Positive          No           Fast Food                     33  
363    Yes   Positive         Yes            Desserts                     35  
369    Yes   Positive         Yes        Healthy Food                     60  
374    Yes  Negative          Yes               Vegan                     58  
386    Yes   Positive         Yes          Vegetarian                     42  

[285 rows x 15 columns]

###**Incosistent Value**

Pada dataset nilai nilai dalam kolom sudah sesuai.

##**CONSTRUCT DATA**

In [75]:
food_types = ["Vegetarian", "Non-Vegetarian", "Vegan", "Fast Food", "Healthy Food", "Desserts"]

df["Preferred Food Type"] = np.random.choice(food_types, size=len(df))

df["Average Delivery Time"] = np.random.randint(20, 61, size=len(df))

print(df.head())

    Age  Gender Marital Status Occupation  Monthly Income  \
0  20.0  Female         Single    Student       No Income   
1  24.0  Female         Single    Student  Below Rs.10000   
2  22.0    Male         Single    Student  Below Rs.10000   
3  22.0  Female         Single    Student       No Income   
4  22.0    Male         Single    Student  Below Rs.10000   

  Educational Qualifications  Family size  latitude  longitude  Pin code  \
0              Post Graduate            4   12.9766    77.5993    560001   
1                   Graduate            3   12.9770    77.5773    560009   
2              Post Graduate            3   12.9551    77.6593    560017   
3                   Graduate            6   12.9473    77.5616    560019   
4              Post Graduate            4   12.9850    77.5533    560010   

  Output   Feedback Unnamed: 12 Preferred Food Type  Average Delivery Time  
0    Yes   Positive         Yes           Fast Food                     51  
1    Yes   Positive   

<ipython-input-75-4baf710555d1>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["Preferred Food Type"] = np.random.choice(food_types, size=len(df))
<ipython-input-75-4baf710555d1>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["Average Delivery Time"] = np.random.randint(20, 61, size=len(df))


##**DATA REDUCTION**

In [77]:
df = df.drop('Unnamed: 12', axis=1)

In [78]:
df

Age  Gender Marital Status Occupation   Monthly Income  \
0    20.0  Female         Single    Student        No Income   
1    24.0  Female         Single    Student   Below Rs.10000   
2    22.0    Male         Single    Student   Below Rs.10000   
3    22.0  Female         Single    Student        No Income   
4    22.0    Male         Single    Student   Below Rs.10000   
..    ...     ...            ...        ...              ...   
355  21.0    Male         Single    Student        No Income   
363  30.5    Male        Married   Employee  More than 50000   
369  30.0    Male        Married   Employee  More than 50000   
374  21.0    Male         Single    Student        No Income   
386  23.0    Male         Single    Student   Below Rs.10000   

    Educational Qualifications  Family size  latitude  longitude  Pin code  \
0                Post Graduate            4   12.9766    77.5993    560001   
1                     Graduate            3   12.9770    77.5773    560009   
2                Post Graduate            3   12.9551    77.6593    560017   
3                     Graduate            6   12.9473    77.5616    560019   
4                Post Graduate            4   12.9850    77.5533    560010   
..                         ...          ...       ...        ...       ...   
355                   Graduate            2   13.0012    77.5995    560046   
363                       Ph.D            3   12.9306    77.5434    560085   
369              Post Graduate            6   12.9369    77.6407    560095   
374                   Graduate            3   13.0641    77.5931    560092   
386              Post Graduate            2   12.9770    77.5773    560009   

    Output   Feedback Preferred Food Type  Average Delivery Time  
0      Yes   Positive           Fast Food                     51  
1      Yes   Positive        Healthy Food                     49  
2      Yes  Negative                Vegan                     50  
3      Yes   Positive               Vegan                     27  
4      Yes   Positive           Fast Food                     55  
..     ...        ...                 ...                    ...  
355     No   Positive           Fast Food                     35  
363    Yes   Positive           Fast Food                     25  
369    Yes   Positive      Non-Vegetarian                     53  
374    Yes  Negative                Vegan                     48  
386    Yes   Positive               Vegan                     31  

[285 rows x 14 columns]